src/espidf.rs

```diff
@@ -360,26 +329,71 @@ impl Installer {
         log::debug!("Using PATH='{}'", &paths.to_string_lossy());
 
         Ok(EspIdf {
-            install_dir,
             esp_idf,
-            esp_idf_version: self.version,
             exported_path: paths,
-            venv_python: python,
         })
     }
 
+    /// Find a possible installed esp-idf git repository.
+    ///
+    /// This will search in te directory location supplied by the caller.
+    fn find_esp_idf(indir: impl AsRef<Path>, gitref: &git::Ref) -> Result<Option<git::Repository>> {
```

Comment:
> I think `impl AsRef<Path>` is pretty useless here and complicates the API. This is only useful with frequently used / utility functions.

I don't have a strong feeling about that. Since I made `find_esp_idf`private,  I'll turn it into `&str` no prob.

> Also, my criticism of the return type also applies here; I think instead of `Result<Option<T>>`, a custom error type would be better.

I'll probably just rename it to `try_find_esp_idf`. As per above, I don't feel `Result<Option>` being so awkward as long as it is reflected in the function name. Moreover, custom error types are a pain. Sure, `thiserror` and so on, but introducing it just for that case, in a codebase that is otherwise completely `anyhow`-ish?

Reply:
Done. The whole conversation is now moot, because `find_esp_idf no longer exists as it was merged with `clone` and moved into `Managed::resolve`. As a result, we no longer have the `Result<Option>` type.

Reply:
Nice 👍